### ISSUE TITLE
Export collection version as zipped JSON

### ISSUE THREAD

**Issue by AnjaliWadhwa** [OpenConceptLab/oclapi#204]
https://github.com/OpenConceptLab/oclapi/wiki/Export-API
- User should be able to export a version of collection.
- Exported file should be saved on S3.
- Add lastUpdated and lastUpdatedTimezone into the response header of GET request.
- Rename the exportURL property in the response header to Location as per the HTTP/REST guidelines
- Task status should be visible in Flower.
- The export filename takes the following form:
  
  `[:repositoryId]_[:repoVersion].[:lastUpdated].tar`

**Comment by AnjaliWadhwa** [comment]
Notes:
* User should not be allowed to export HEAD. created a separate bug for it #205 to fix for both collection and source.
* Export can work for any valid version of collection.
* User should be able to export the latest version of collection by 'latest' endpoint?

**Comment by paynejd** [comment]
Should collection export also contain the references?

For example:
```
{
  "type": "Collection",
  "id": "my-collection",
  "name": "My Collection",
  "concepts": [
    # concepts here ...
  ],
  "mappings": [
    # mappings here ...
  ],
  "references": [
    # references here ...
  ]
}
```

**Comment by snyaggarwal** [comment]
As of now they do.

**Comment by AnjaliWadhwa** [comment]
This is working fine. Only issue is that the metadata in export comes from Collection instead of version. It will be fixed in #163

**Comment by AnjaliWadhwa** [comment]
Below is the content of export file for collection version.

```
{  
   "type":"Collection",
   "uuid":"57cea2790f9d1100147f1c6a",
   "id":"WHO-testMlcaE",
   "short_code":"WHO-testMlcaE",
   "name":"Woreda Health Office",
   "full_name":"Woreda Health Office Collection",
   "description":"",
   "external_id":"",
   "collection_type":"Dictionary",
   "public_access":"View",
   "default_locale":"ar",
   "supported_locales":[  
      "en",
      "es"
   ],
   "website":"",
   "url":"/orgs/EthiopiaMoH-test-MlcaE/collections/WHO-testMlcaE/",
   "versions_url":"/orgs/EthiopiaMoH-test-MlcaE/collections/WHO-testMlcaE/versions/",
   "concepts_url":"/orgs/EthiopiaMoH-test-MlcaE/collections/WHO-testMlcaE/concepts/",
   "mappings_url":"/orgs/EthiopiaMoH-test-MlcaE/collections/WHO-testMlcaE/mappings/",
   "owner":"EthiopiaMoH-test-MlcaE",
   "owner_type":"Organization",
   "owner_url":"/orgs/EthiopiaMoH-test-MlcaE/",
   "versions":6,
   "created_on":"2016-09-06T07:03:21.834",
   "updated_on":"2016-09-08T07:21:07.491",
   "created_by":"awadhwa1",
   "updated_by":"awadhwa1",
   "extras":null,
   "references":[  
      {  
         "expression":"/orgs/EthiopiaMoH-test-MlcaE/sources/HSTP-Indicators/mappings/57cea2770f9d1100147f1c69/",
         "reference_type":"mappings"
      },
      {  
         "expression":"/orgs/EthiopiaMoH-test-MlcaE/sources/HSTP-Indicators/concepts/C1.1.1.2-MlcaE/",
         "reference_type":"concepts"
      }
   ],
   "concepts":[  
      {  
         "type":"Concept",
         "uuid":"57cea2730f9d1100147f1c68",
         "id":"C1.1.1.2-MlcaE",
         "external_id":"",
         "concept_class":"Aggregate Measurement",
         "datatype":"None",
         "display_name":"Life Expectancy at Birth (years)",
         "display_locale":"ar",
         "names":[  
            {  
               "uuid":"b784eb02c820487c91dac8e6c2b57eee",
               "external_id":null,
               "name":"Life Expectancy at Birth (years)",
               "locale":"ar",
               "locale_preferred":false,
               "name_type":"Fully Specified",
               "type":"ConceptName"
            }
         ],
         "descriptions":[  
            {  
               "uuid":"b37f1eb7bd264f5e809e425fab09e3a4",
               "external_id":null,
               "description":"",
               "locale":"ar",
               "locale_preferred":false,
               "description_type":"",
               "type":"ConceptDescription"
            }
         ],
         "retired":false,
         "source":"HSTP-Indicators",
         "source_url":"/orgs/EthiopiaMoH-test-MlcaE/sources/HSTP-Indicators/",
         "owner":"EthiopiaMoH-test-MlcaE",
         "owner_type":"Organization",
         "owner_url":"/orgs/EthiopiaMoH-test-MlcaE/",
         "version":"57cea2730f9d1100147f1c68",
         "created_on":"2016-09-06T07:03:15.519",
         "updated_on":"2016-09-06T07:03:15.838",
         "version_created_on":"2016-09-06T07:03:15.519",
         "version_created_by":"awadhwa1",
         "extras":null,
         "mappings":[  

         ],
         "version_url":"/orgs/EthiopiaMoH-test-MlcaE/sources/HSTP-Indicators/concepts/C1.1.1.2-MlcaE/57cea2730f9d1100147f1c68/",
         "url":"/orgs/EthiopiaMoH-test-MlcaE/sources/HSTP-Indicators/concepts/C1.1.1.2-MlcaE/"
      }
   ],
   "mappings":[  
      {  
         "type":"Mapping",
         "id":"57cea2770f9d1100147f1c69",
         "external_id":"",
         "retired":false,
         "map_type":"SAME-AS",
         "from_source_owner":"EthiopiaMoH-test-MlcaE",
         "from_source_owner_type":"Organization",
         "from_source_name":"HSTP-Indicators",
         "from_source_url":null,
         "from_concept_code":"C1.1.1.2-",
         "from_concept_name":"Life Expectancy at Birth (years)",
         "from_concept_url":"/orgs/EthiopiaMoH-test-MlcaE/sources/HSTP-Indicators/concepts/C1.1.1.2-/",
         "to_source_owner":"EthiopiaMoH-test-MlcaE",
         "to_source_owner_type":"Organization",
         "to_source_name":"HSTP-Indicators",
         "to_source_url":"/orgs/EthiopiaMoH-test-MlcaE/sources/HSTP-Indicators/",
         "to_concept_code":"C1.1.1.2-MlcaE",
         "to_concept_name":"Life Expectancy at Birth (years)",
         "to_concept_url":"/orgs/EthiopiaMoH-test-MlcaE/sources/HSTP-Indicators/concepts/C1.1.1.2-MlcaE/",
         "source":"HSTP-Indicators",
         "owner":"EthiopiaMoH-test-MlcaE",
         "owner_type":"Organization",
         "url":"/orgs/EthiopiaMoH-test-MlcaE/sources/HSTP-Indicators/mappings/57cea2770f9d1100147f1c69/",
         "extras":null,
         "created_at":"2016-09-06T07:03:19.007",
         "updated_at":"2016-09-06T07:03:19.164",
         "created_by":"awadhwa1",
         "updated_by":"awadhwa1"
      }
   ]
}
```

**Comment by paynejd** [comment]
The collection export file does not match what is described above. Please see #236

**Comment by paynejd** [comment]
To confirm, here is the expected structure for the collection export file:
```
{
    "type": "Collection",
    .... # collection details from the time that the version was created
    "collection_version": {
        ... # collection version details
    },
    "concepts": [ ... ],
    "mappings": [ ... ],
    "references": [ ... ]
}
```

Also to confirm, here is the expected structure for the **source** export file (same as collections above, but without the references):
```
{
    "type": "Source",
    .... # source details from the time that the version was created
    "source_version": {
        ... # source version details
    },
    "concepts": [ ... ],
    "mappings": [ ... ]
}
```

**Comment by AnjaliWadhwa** [comment]
This is being fixed in #236
